test/unit/controllers/declaration/CusCodeControllerSpec.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.CusCodeController
+import forms.declaration.CusCode
+import forms.declaration.CusCode._
+import models.DeclarationType.DeclarationType
+import models.{DeclarationType, Mode}
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
+import play.api.data.Form
+import play.api.libs.json.{JsObject, JsString}
+import play.api.mvc.Call
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.cus_code
+
+class CusCodeControllerSpec extends ControllerSpec {
+
+  val mockPage = mock[cus_code]
+
+  val controller =
+    new CusCodeController(mockAuthAction, mockJourneyAction, mockExportsCacheService, navigator, stubMessagesControllerComponents(), mockPage)(ec)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(mockPage.apply(any(), any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    reset(mockPage)
+  }
+
+  val itemId = "itemId"
+
+  def theResponseForm: Form[CusCode] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[CusCode]])
+    verify(mockPage).apply(any(), any(), captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  private def formData(code: String) = JsObject(Map(cusCodeKey -> JsString(code), hasCusCodeKey -> JsString("Yes")))
+
+  "CUSCode controller" must {
+
+    onEveryDeclarationJourney() { declaration =>
+      "return 200 (OK)" when {
+
+        "display page method is invoked and cache is empty" in {
+
+          withNewCaching(declaration)
+
+          val result = controller.displayPage(Mode.Normal, itemId)(getRequest())
+
+          status(result) mustBe OK
+          verify(mockPage, times(1)).apply(any(), any(), any())(any(), any())
+
+          theResponseForm.value mustBe empty
+        }
+
+        "display page method is invoked and cache contains data" in {
+          val cusCode = CusCode(Some("12345678"))
+          val item = anItem(withCUSCode(cusCode))
+          withNewCaching(aDeclarationAfter(declaration, withItems(item)))
```

Comment:
What is `aDeclarationAfter`?

Reply:
That function is to get a declaration after applying modifiers - introduce while ago for easier manipulating declaration and build it incrementally

Reply:
IMO the name is wrong and don't give any value, IMO `aDeclaration` is enough, word after didn't give anything for me